src/wp-includes/class-wp-theme-json.php

```diff
@@ -1925,6 +1925,48 @@ public function get_styles_block_nodes() {
 		return static::get_block_nodes( $this->theme_json );
 	}
 
+	/**
+	 * Returns a filtered declarations array if there is a separator block with only a background
+	 * style defined in theme.json by adding a color attribute to reflect the changes in the front.
+	 *
+	 * @since 6.1.1
+	 *
+	 * @param array $declarations List of declarations.
+	 * @return array $declarations List of declarations filtered.
+	 */
+	private static function update_separator_declarations( $declarations ) {
+		$background_matches = array_values(
+			array_filter(
+				$declarations,
+				static function( $declaration ) {
+					return 'background-color' === $declaration['name'];
+				}
+			)
+		);
+
+		if ( isset( $background_matches[0]['value'] ) ) {
+			$border_color_matches = false;
+			$text_color_matches   = false;
+
+			foreach ( $declarations as $declaration ) {
+				if ( 'border-color' === $declaration['name'] ) {
+					$border_color_matches = true;
+				} elseif ( 'color' === $declaration['name'] ) {
+					$text_color_matches = true;
+				}
+			}
```

Comment:
Now that I look at the code, I see what you were saying @dream-encode:

In the `foreach`, once it finds a single `true, then it can break out of the `foreach`. 

Why? The next conditional is checking if both are `false`. As soon as one of the variables is `true`, then that conditional will not execution. So no need to keep iterating in the `foreach`.

My suggestion:

Add a `break` to each:

```
foreach ( $declarations as $declaration ) {
	if ( 'border-color' === $declaration['name'] ) {
		$border_color_matches = true;
		break;
	} elseif ( 'color' === $declaration['name'] ) {
		$text_color_matches = true;
		break;
	}
}
```

What do you think @dream-encode ?

Reply:
I think a break in both would not work properly since we need to absolutely determine if both are true. Line 1959 check both.  If we bail on the first one found, the other may yet still exist in the array.  However, @felixarntz suggestion to check if both have already been found then break earlier is good.